build/nuke/Build.Steps.cs

```diff
@@ -350,40 +351,123 @@ partial class Build
+                        CopyNativeDependenciesToStore(file, depsJsonContent);
+
+                        RemoveOpenTelemetryAutoInstrumentationAdditionalDepsFromDepsFile(depsJsonContent, file);
+                    });
+                RemoveFilesFromAdditionalDepsDirectory();
+
+                void CopyNativeDependenciesToStore(AbsolutePath file, string depsJsonContent)
+                {
+                    var depsDirectory = file.Parent;
+                    var targetDirectory = Path.Combine(depsDirectory.Parent.Parent.Parent.Parent, "store");
+                    using var jsonDocument = JsonDocument.Parse(depsJsonContent);
+
+                    var runtimeName = jsonDocument.RootElement.GetProperty("runtimeTarget").GetProperty("name").GetString();
+                    var folderRuntimeName = MapToFolderName(runtimeName);
+
+                    string MapToFolderName(string runtimeName)
+                    {
+                        switch (runtimeName)
+                        {
+                            case ".NETCoreApp,Version=v3.1":
+                                return "netcoreapp3.1";
+                            case ".NETCoreApp,Version=v6.0":
+                                return "net6.0";
+                        }
+
+                        throw new ArgumentOutOfRangeException(nameof(runtimeName), runtimeName,
+                            "This value is not supported. You have probably introduced new .NET version to AutoInstrumentation");
+                    }
+
+                    foreach (var targetProperty in jsonDocument.RootElement.GetProperty("targets").EnumerateObject())
+                    {
+                        var target = targetProperty.Value;
+
+                        foreach (var packages in target.EnumerateObject())
+                        {
+                            if (packages.Value.TryGetProperty("runtimeTargets", out var runtimeTargets))
+                            {
+                                foreach (var runtimeDependency in runtimeTargets.EnumerateObject())
+                                {
+                                    var sourceFileName = Path.Combine(depsDirectory, runtimeDependency.Name);
+
+
+                                    var targetFileNameX64 = Path.Combine(targetDirectory, "x64", folderRuntimeName,
+                                        packages.Name.ToLowerInvariant(), runtimeDependency.Name);
+                                    var targetFileNameX86 = Path.Combine(targetDirectory, "x86", folderRuntimeName,
+                                        packages.Name.ToLowerInvariant(), runtimeDependency.Name);
+
+                                    var targetDirectoryX64 = Path.GetDirectoryName(targetFileNameX64);
+                                    var targetDirectoryX86 = Path.GetDirectoryName(targetFileNameX86);
+
+                                    if (!Directory.Exists(targetDirectoryX64))
```

Comment:
nit: could be removed to simplify code
See remarks at https://learn.microsoft.com/en-us/dotnet/api/system.io.directory.createdirectory?view=net-7.0